components/usage/pkg/apiv1/usage.go

```diff
@@ -224,6 +222,8 @@ func (s *UsageService) ReconcileUsageWithLedger(ctx context.Context, req *v1.Rec
 	logger.Infof("Found %d workspaces instances for usage records in draft.", len(instancesWithUsageInDraft))
 	instances = append(instances, instancesWithUsageInDraft...)
 
+	// now has to be computed after we've collected all data, to ensure that it's always greater than any of the records we fetch
+	now := s.nowFunc()
```

Comment:
I think this might be an improvement and will fix ti.

:cloud: Note that this alone is not a guarantee that startTimes are not greated than now in a distributed setting. But maybe reasonably safe to assume, yes.